test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go

```diff
@@ -0,0 +1,302 @@
+}
+
+// SetupSuite waits for both fakeintakes to be ready before running tests.
+func (v *multiFakeIntakeSuite) SetupSuite() {
+	v.Env() // update the environment outside EventuallyWithT
+
+	// Wait for the fakeintake to be ready to avoid 503
+	require.EventuallyWithT(v.T(), func(c *assert.CollectT) {
+		assert.NoError(c, v.Env().Fakeintake1.Client.GetServerHealth())
+		assert.NoError(c, v.Env().Fakeintake2.Client.GetServerHealth())
+	}, intakeMaxWaitTime, intakeTick)
+}
+
+// BeforeTest flushes both fakeintakes before starting each test.
+func (v *multiFakeIntakeSuite) SetupTest() {
+	v.NoError(v.Env().Fakeintake1.FlushServerAndResetAggregators())
+	v.NoError(v.Env().Fakeintake2.FlushServerAndResetAggregators())
+}
+
+// TestNSSFailover tests that the agent correctly picks-up an NSS change of the intake.
+//
+// The test uses two fakeintakes to represent two backends, and the /etc/hosts file for the NSS source,
+// setting-up an entry for the intake, pointing to the first fakeintake, then changing that entry
+// to point to the second fakeintake without restarting the agent.
+//
+// The test checks that metrics, logs, and flares are properly received by the first intake before
+// the failover, and by the second one after.
+//
+// Note: although the man page of `nsswitch.conf` states that each process using it should only read
+// it once (ie. no reload), glibc and Go reload it periodically
+// cf. https://go-review.googlesource.com/c/go/+/448075
+//
+// TODO: handle APM traces
+func (v *multiFakeIntakeSuite) TestNSSFailover() {
+	// ensure host uses files for NSS
+	enforceNSSwitchFiles(v.T(), v.Env().VM)
+
+	// setup NSS entry for intake
+	fakeintake1IP, err := hostIPFromURL(v.Env().Fakeintake1.URL())
+	v.NoError(err)
+	setHostEntry(v.T(), v.Env().VM, intakeName, fakeintake1IP)
+
+	// configure agent to use the custom intake, set connection_reset_interval, use logs, and processes
+	agentOptions := []agentparams.Option{
+		agentparams.WithAgentConfig(getAgentConfig(connectionResetInterval)),
+		agentparams.WithLogs(),
+		agentparams.WithIntakeHostname(intakeName),
+		agentparams.WithIntegration("custom_logs.d", customLogsConfig),
+	}
+	v.UpdateEnv(multiFakeintakeStackDef(agentOptions...))
+
+	// check that fakeintake1 does receive metrics
+	v.T().Logf("checking that the agent contacts main intake at %s", fakeintake1IP)
+	require.EventuallyWithT(
+		v.T(),
+		assertIntakeIsUsed(v.Env().VM, v.Env().Fakeintake1, v.Env().Agent),
+		intakeMaxWaitTime,
+		intakeTick,
+	)
+
+	// check that fakeintake2 doesn't receive metrics
+	assertIntakeNotUsed(v.T(), v.Env().VM, v.Env().Fakeintake2, v.Env().Agent)
+
+	// perform NSS change
+	fakeintake2IP, err := hostIPFromURL(v.Env().Fakeintake2.URL())
+	v.NoError(err)
+	setHostEntry(v.T(), v.Env().VM, intakeName, fakeintake2IP)
+
+	// check that fakeintake2 receives metrics
+	v.T().Logf("checking that the agent contacts fallback intake at %s", fakeintake2IP)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You could flush the first one and assert it does not receive any other metric

Reply:
I wasn't sure how to do that properly because if I flush and directly check then it doesn't really makes sense
I guess I could flush, sleep, and check ?

Reply:
> I guess I could flush, sleep, and check ?

Did that